### ISSUE TITLE
getPageSource() bug! (facebook webdriver(php) - firefox)

### ISSUE THREAD

**Issue by mihutz** [SeleniumHQ/selenium#1344]
When trying to get page source after navigating to an url which contains "#"(bookmark) I recieve the last source before the current get().

Example: 
$webdriver->get('http://www.domain1.com');
echo md5($webdriver->getPageSource()); // source1 
$webdriver->get('http://www.domain1.com/anotherRoute#test');
echo md5($webdriver->getPageSource()); // source2 == source1

If I change the domain:
$webdriver->get('http://www.domain1.com');
echo md5($webdriver->getPageSource()); // source1 
$webdriver->get('http://www.domain2.com/anotherRoute#test');
echo md5($webdriver->getPageSource()); // source1 != source2

I made some test locally and the webbrowser navigated correctly. Only the html sources got by selenium are the same.

**Comment by ddavison** [comment]
have you brought this up with [Facebook's PHPWebDriver](https://github.com/facebook/php-webdriver/issues)?

**Comment by mihutz** [comment]
Hi @ddavison , yes, I used Facebook's PHPWebDriver. No, there is no ajax, just the #bookmark which will scroll down the page. I investigate more into this and I found that selenium doesn't wait for the browser to complete the loading if you try to navigate to an url from the same domain if there are any bookmarks(it just send the orders to the browser without any waiting). When you navigate to another url , then everything is fine and selenium waits for the browser to finish the loading for the first page and navigate to the next page only after loading is completed.

Thanks,
Claudiu
